packages/headless/src/controllers/commerce/core/pagination/headless-core-commerce-pagination.ts

```diff
@@ -170,6 +177,10 @@ export function buildCorePagination(
       dispatch(props.fetchProductsActionCreator());
     },
 
+    setPageSizeWithoutFetch(pageSize: number) {
+      dispatch(setPageSize({slotId, pageSize}));
+    },
+
```

Comment:
I am unsure what to do here. The problem is that `setPageSize` (expand the file to see it just above) sets the page size and ALSO fetches products via `dispatch(props.fetchProductsActionCreator());`. This causes a problem in atomic because `setPageSize` is used to set the `initialChoice` value. On the first load, a second search is triggered at the same time of the first (only one with the correct pageSize value) which causes big problems with the url manager and race conditions. 

Two options :
 - Add a new `setPageSizeWithoutFetch` that only sets the headless state value without fetching products.
 - Remove the  `dispatch(props.fetchProductsActionCreator());` from the setPageSize and manually do `setPageSize` + `fetchMoreProducts` in Atomic. (Possibly breaking on the headless side)

Reply:
I could also call it `SetInitialPageSize`

Reply:
Another option would be to dispatch the `setPageSize` action directly in Atomic to set the `initialChoice` value.

Reply:
No, that is precisely the problem. In the `initialize` function, we set the intialChoice using setPageSize which triggers a second search. We need a option to setPageSize without triggering a search for this use case in the initialize function that happens before the `executeFirstRequest` is triggered.

<img width="455" alt="image" src="https://github.com/user-attachments/assets/bdde58e4-4515-4a9b-9963-57ca42fa6ff1">

Reply:
<img width="397" alt="image" src="https://github.com/user-attachments/assets/04f53eca-50b7-4ffc-b73c-861bc513ac62">

Reply:
my mistake, I can just enginge.dispatch 😅